### PR TITLE
WIP - Added ViewLayer components

### PR DESCRIPTION
**Pull request by bill-luu**
Copied and pasted files from https://github.com/UCSolarCarTeam/Epsilon-Desktop-Client-Telemetry and made changes to the appropriate `.pro` files.

### CODE REVIEW

**Comment by bill-luu** [comment]
Hey @enochtsang before you review this, I'm getting a strange `undefined reference to ViewContainer::ViewContainer()` error, which is strange and giving me aids since I'm certain that I have all the includes and `.pro`s set up right. Can you take a look and see what the issue is?

**Comment by ExtraGravity** [comment]
The command line args should be set up in this PR. I will take a look at the problem.

File: ViewLayer/ViewContainer.h [hunk 1]
```diff
@@ -4,10 +4,12 @@
 class TelemetryDisplayUi;
 class OverlordWidget;
 
+enum class Mode { HERMES, TELEMETRY };
```
Comment (ExtraGravity) [inline ViewLayer/ViewContainer.h]: I'd prefer `HEADLESS, GUI`

**Review by ExtraGravity — CHANGES_REQUESTED** [review]

**Comment by ExtraGravity** [comment]
I'll test this when I get home, but code changes look good. You need to format the repo though.

**Comment by ExtraGravity** [comment]
I would expect some sort of GUI to come up with the flag, even in an error state. This is not the case currently.

**Comment by bill-luu** [comment]
I could add `resets` for the Business and the Communication container after the `viewContainer` one, but I would imagine that the ViewLayer will need access to those layers. Should I do something along the lines of  
`viewLayer_.reset(new ViewContainer());`
`businessContainer_.reset(new BusinessContainer());`
`viewLayer_.addBusinessContainer(*BusinessContainer)`

**Comment by ExtraGravity** [comment]
To reiterate from our FB chat, Comm Layer should loop in error if it's headless mode, if it's headless mode it won't even try to connect without response from the GUI.

**Comment by bill-luu** [comment]
Is there a better way of including `Mode.h`? I feel like this is pretty janky right now. I'm unable to forward declare it, so I'm wondering if there are alternatives to forward declaring or if I'm just doing it wrong. 

Also, I noticed that not just the comms layer needs to be changed, for instance the business layer shouldn't be forwarding stuff in `GUI` mode. I'll have to look into the extent of what needs to be changed.

**Comment by ExtraGravity** [comment]
I agree it's weird to have Mode.h everywhere. That should probably be abstracted acutally, so for example instead of passing the Mode into ViewLayer, the ViewLayer just doens't get instantiated if the Mode is headless. Another example is UdpMessageForwarder just takes in an argument of True or False on whether to initiate a connection right away or not. So we try not to propogate the mode down, but instead of make decisions as early on as we can.

**Comment by ExtraGravity** [comment]
New plan for desktop.

Run Hermes to convert serial to json, send to localhost. Run webserver and access webpage via localhost.

**Pull request closed without merging**